### PR TITLE
react-tinacms-github: improve error handling around 404s

### PR DESCRIPTION
**Pull request by ncphillips**
This case checks all the reasons there may have been a 404:
- Unauthorized
  - Private Repo => Cancel
  - Missing Fork => Create Fork or Cancel
- Missing Branch => Switch to base branch (i.e. master)
- Missing File => Switch to base branch

Each of those have their own custom text

### CODE REVIEW

File: packages/react-tinacms-github/src/github-error/github-interpeter.ts [hunk 1]
```diff
@@ -46,24 +59,57 @@ export const getModalProps = async (
...
+
+      // Is the user authorized to edit this repo?
+      if (!(await githubClient.isAuthorized())) {
+        // Is the repo public?
+        if (githubClient.authScope === 'public_repo') {
+          return {
+            title: `Create a Fork of ${githubClient.baseRepoFullName}`,
+            message:
+              `You do not have permission to make changes to ${githubClient.baseRepoFullName}.` +
+              `Press the button below to fork this site and begi editing. `,
```
Comment (dwalkr) [inline packages/react-tinacms-github/src/github-error/github-interpeter.ts:79]:
```suggestion
              `Press the button below to fork this site and begin editing. `,
```

File: packages/react-tinacms-github/src/github-error/github-interpeter.ts [hunk 2]
```diff
@@ -46,24 +59,57 @@ export const getModalProps = async (
...
       }
+
+      // The file is missing
       return {
-        title: '404 Not Found',
-        message: 'You are missing a fork.',
-        actions: [cancelEditModeAction, reauthenticateAction],
+        title: 'Content Not Found.',
+        message:
+          'The file your are trying to access is missing. Maybe it lives on a different branch?',
```
Comment (dwalkr) [inline packages/react-tinacms-github/src/github-error/github-interpeter.ts:111]:
```suggestion
          'The file you are trying to access is missing. Maybe it lives on a different branch?',
```

**Comment by kendallstrautman** [comment]
@all-contributors please add @omarcostahamido for docs and bugs

**Comment by omarcostahamido** [comment]
what!?
oh man... ❤️
thank you very much :)

**Pull request merged**